### DISCUSSION TITLE
Expose Cron job list

### DISCUSSION THREAD

**Discussion by mlusetti** [pocketbase/pocketbase#6046] [Q&A] [answered]
First of all congrats to you for the great work on 0.23 ... PB it's even better then before (from an integration point of view)

I'm in the situation where I've my own cron schedules defined in the DB.
Now I'm studying the best solution to have it translated within app.Cron() and I'm finding myself in the situation where it would be nice to have jobs from Cron struct exposed, I mean the list or the map. Even in a read only form.

Curious why it's not published it yet.
Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
The jobs map is not public because there was no need for it.

There is [task in the roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=89761221) to allow listing and manually triggering the registered app level cron jobs from the UI but there are no ETAs when that will be implemented.

**Reply by mlusetti** [reply]
Thank for the reply.

Would you accept a PR to list just name job names (for now) ?
I mean for my use case would be enough to have names of the job.

BTW now the removeAll API will remove ALL cron jobs... even the system defined ones. Do you think this will remain the case ?

Thanks!

**Reply by ganigeorgiev** [reply]
> Would you accept a PR to list just name job names (for now) ?

No.

> TW now the removeAll API will remove ALL cron jobs... even the system defined ones. Do you think this will remain the case ?

Yes. It is is similar to the [`app.Store()`](https://pocketbase.io/docs/go-miscellaneous/#app-store). I'll consider in the future adding a similar warning or maybe change `app.Cron()` to return an interface with subset of the original struct methods.

**Reply by ganigeorgiev** [reply]
_Just for info in case someone stumble here, `app.Cron().Jobs()` helper method that returns a shallow copy of the registered jobs slice is available in the `develop` branch._
